### ISSUE TITLE
Inconsistent demo results

### ISSUE THREAD

**Issue by gndlwch2w** [jeromerony/distorch#7]
Dear developer, thank you very much for your work. I installed the library you provided into my environment and tested the demo you provided, and found that **some indicators were calculated inconsistently**.

I execute the given demo in my environment as follows:
```python
import torch
import distorch
device = torch.device('cuda')
A = torch.tensor([[0, 0, 0, 0, 0],
                  [0, 1, 0, 0, 0],
                  [0, 0, 1, 1, 0],
                  [0, 1, 1, 1, 0],
                  [0, 0, 0, 0, 0]], dtype=torch.bool, device=device)
B = torch.tensor([[1, 1, 0, 0, 0],
                  [0, 1, 1, 0, 0],
                  [0, 1, 1, 1, 0],
                  [0, 1, 1, 0, 0],
                  [0, 0, 0, 0, 0]], dtype=torch.bool, device=device)
metrics = distorch.boundary_metrics(A, B)
print(metrics)

# My outputs
DistanceMetrics(Hausdorff=tensor(1.4142, device='cuda:0'), 
                Hausdorff95_1_to_2=tensor(1., device='cuda:0'), 
                Hausdorff95_2_to_1=tensor(1.1036, device='cuda:0'), 
                AverageSurfaceDistance_1_to_2=tensor(0.2857, device='cuda:0'), 
                AverageSurfaceDistance_2_to_1=tensor(0.4009, device='cuda:0'), 
                AverageSymmetricSurfaceDistance=tensor(0.3471, device='cuda:0'), 
                NormalizedSurfaceDistance_1_to_2=tensor(1., device='cuda:0'), 
                NormalizedSurfaceDistance_2_to_1=tensor(0.9375, device='cuda:0'), 
                NormalizedSymmetricSurfaceDistance=tensor(0.9667, device='cuda:0'))

# Your outputs
DistanceMetrics(Hausdorff=tensor(1.4142, device='cuda:0'),
                Hausdorff95_1_to_2=tensor(1., device='cuda:0'),
                Hausdorff95_2_to_1=tensor(1.1036, device='cuda:0'),
                AverageSurfaceDistance_1_to_2=tensor(0.2308, device='cuda:0'),
                AverageSurfaceDistance_2_to_1=tensor(0.4009, device='cuda:0'),
                AverageSymmetricSurfaceDistance=tensor(0.3246, device='cuda:0'),
                NormalizedSurfaceDistance_1_to_2=tensor(1., device='cuda:0'),
                NormalizedSurfaceDistance_2_to_1=tensor(0.9375, device='cuda:0'),
                NormalizedSymmetricSurfaceDistance=tensor(0.9655, device='cuda:0'))
```
My environment configuration:
```bash
Ubuntu 24.04.2 LTS
Python 3.11.13
Torch 2.0.1
pytorch-cuda=11.8
Driver Version: 575.64.03
CUDA Version: 12.9
Distorch 0.1.2 (Triton)
```
In addition, since torch 2.0.1 does not support `any(tensor, dim=tuple)` API, I changed the source code as follows:
```python

# Before
non_empty = masks[0].any(dim=other_dims).logical_or_(masks[1].any(dim=other_dims))

# After
def any_dims(x: torch.Tensor, dims, keepdim=False) -> torch.Tensor:
    if isinstance(dims, int):
        return torch.any(x, dim=dims, keepdim=keepdim)
    dims = sorted(dims, reverse=True)
    for d in dims:
        x = torch.any(x, dim=d, keepdim=keepdim)
    return x

non_empty = any_dims(masks[0], dims=other_dims).logical_or_(any_dims(masks[1], dims=other_dims))
```
Thank you very much for your attention!

**Comment by jeromerony** [comment]
Closing for inactivity.
